pkg/certificate/providers/config.go

```diff
@@ -154,6 +154,45 @@ func (c *Config) GetCertificateManager() (certificate.Manager, debugger.Certific
 	}
 }
 
+// SynchronizeCertificate is a helper function that ensures creation and synchronization of a certificate
+// using Kubernetes Secrets backend and API atomicity.
+func SynchronizeCertificate(ns string, secretName string, cert certificate.Certificater, kubeClient kubernetes.Interface) (certificate.Certificater, error) {
+	// Attempt to create it in K8s. When creating, only one of the creates will succeed. All others will get
+	secretData := map[string][]byte{
+		constants.KubernetesOpaqueSecretCAKey:             cert.GetCertificateChain(),
+		constants.KubernetesOpaqueSecretCAExpiration:      []byte(cert.GetExpiration().Format(constants.TimeDateLayout)),
+		constants.KubernetesOpaqueSecretRootPrivateKeyKey: cert.GetPrivateKey(),
+	}
+
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      secretName,
+			Namespace: ns,
+		},
+		Data: secretData,
+	}
+
+	_, err := kubeClient.CoreV1().Secrets(ns).Create(context.TODO(), secret, metav1.CreateOptions{})
+
+	if err == nil {
```

Comment:
We could merge these 3 lines since no variables need to be exposed outside this scope.